File: server/database/scheam/banner.js
import mongoose from 'mongoose'
const Schema = mongoose.Schema
const Mixed = Schema.Types.Mixed
const ObjectId = Schema.Types.ObjectId

/**
 * 首页轮播集合
 */
const BannerSchema = new Schema({
    title: String,
    img: String,
    products: [{
        type: ObjectId,
        ref: 'Product'
    }],
    meta: {
        createdAt: {
            type: Date,
            default: Date.now()
        },
        updatedAt: {
            type: Date,
            default: Date.now()
        }
    }
})

BannerSchema.pre('save', function (next) {
    if (this.isNew) {
        this.meta.createdAt = this.meta.updatedAt = Date.now()
    } else {
        this.meta.updatedAt = Date.now()
    }

    next()
})

mongoose.model('Banner', BannerSchema)

File: minappServer/minappServer.Repositories/Entities/Banner.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace minappServer.Repositories.Entities
{
    public class Banner
    {
        public Banner()
        {
            products = new List<string>();
        }
        public string Title { get; set; }
        public string Img { get; set; }
        public List<String> products { get; set; }
    }
}
